### PR TITLE
👷‍♂️ Remove terser for closure compiler

### PR DESCRIPTION
**Pull request by chuckcarpenter**
This branch adds the [rollup-plugin-closure-compiler](https://github.com/ampproject/rollup-plugin-closure-compiler) which reduces the file size a small bit. 

```bash
 Destination: dist/js/shepherd.min.js   │
│   Bundle Size:  49.35 KB                 │
│   Minified Size:  49.29 KB               │
│   Gzipped Size:  15.35 KB
```

after

```bash
 Destination: dist/js/shepherd.min.js   │
│   Bundle Size:  48.8 KB                  │
│   Minified Size:  48.43 KB               │
│   Gzipped Size:  14.89 KB
```

Also, Cypress is run in Firefox as well, for additional coverage. 

🍡

### CODE REVIEW

**Comment by chuckcarpenter** [comment]
how do we validate `compiled code works in IE` now? Just local click tests on a VM?

**Comment by rwwagner90** [comment]
> how do we validate `compiled code works in IE` now? Just local click tests on a VM?

Yeah, just manual testing.

File: package.json [hunk 1]
```diff
@@ -51,6 +54,7 @@
   "module": "dist/js/shepherd.esm.js",
   "dependencies": {
     "@popperjs/core": "^2.1.1",
+    "@ampproject/rollup-plugin-closure-compiler": "^0.24.0",
```
Comment (rwwagner90) [inline package.json]: Shouldn't this be a devDep?

**Comment by rwwagner90** [comment]
@chuckcarpenter just one change request, then I think we're good 👍

**Pull request merged**